### PR TITLE
Allow compilation on OpenBSD

### PR DESCRIPTION
**Pull request by ledeuns**
Two changes to allow to build pgx on OpenBSD :
- pgx requires GNU `make`. Base `make` is BSD `make` so we need to pass `gmake` (via an envvar)
- OpenBSD has `sigsetjmp`

### CODE REVIEW

File: pgx-pg-sys/build.rs [hunk 1]
```diff
@@ -559,8 +559,9 @@ fn build_shim_for_version(
         .unwrap();
     }
 
+    let make = env!("MAKE").unwrap_or("make".to_string());
```
Comment (Hoverbear) [inline pgx-pg-sys/build.rs:562]: In this case `option_env!` is preferable, if `MAKE` is not set with `env!` the build will fail.

Docs: [`env`](https://doc.rust-lang.org/std/macro.env.html) vs [`option_env`](https://doc.rust-lang.org/std/macro.option_env.html)

Comment (ledeuns) [reply pgx-pg-sys/build.rs]: Thank you for the pointer

**Comment by JamesGuthrie** [comment]
FYI If you're planning on using schema generation on OpenBSD then you're probably going to need to add it as a target to https://github.com/zombodb/pgx/blob/master/cargo-pgx/src/templates/cargo_config. See #362 and #340 for more context.

**Comment by eeeebbbbrrrr** [comment]
We're gonna ship this but don't have a way to test.

**Comment by ledeuns** [comment]
I can confirm it works on OpenBSD 7.1. Thank you !

**Comment by eeeebbbbrrrr** [comment]
Amazing! Thanks a bunch for testing and of course this PR. I’m glad we made the changes that accidentally got it working!

**Pull request merged**